packages/enclave-contracts/contracts/slashing/SlashingManager.sol

```diff
@@ -0,0 +1,370 @@
+
+        emit SlashProposed(
+            proposalId,
+            operator,
+            reason,
+            policy.ticketPenalty,
+            policy.licensePenalty,
+            executableAt,
+            msg.sender
+        );
+
+        totalProposals = proposalId + 1;
+    }
+
+    /// @inheritdoc ISlashingManager
+    function executeSlash(uint256 proposalId) external onlySlasher {
+        require(proposalId < totalProposals, InvalidProposal());
+        SlashProposal storage p = _proposals[proposalId];
+
+        // Has already been executed?
+        require(!p.executed, AlreadyExecuted());
+        p.executed = true;
+
+        SlashPolicy memory policy = slashPolicies[p.reason];
+
+        if (policy.requiresProof) {
+            // Appeal window is 0 by policy validation, so we dont check for appeal gating
+            require(p.proofVerified, InvalidProof());
+        } else {
+            // Evidence mode with appeals
+            require(block.timestamp >= p.executableAt, AppealWindowActive());
+            if (p.appealed) {
+                require(p.resolved, AppealPending());
+                require(!p.appealUpheld, AppealUpheld()); // approved = appeal upheld => cancel slash, return?
+            }
+        }
+
+        if (p.ticketAmount > 0) {
+            bondingRegistry.slashTicketBalance(
+                p.operator,
+                p.ticketAmount,
+                p.reason
+            );
+        }
+
+        if (p.licenseAmount > 0) {
+            bondingRegistry.slashLicenseBond(
+                p.operator,
+                p.licenseAmount,
+                p.reason
+            );
+        }
+
+        if (policy.banNode) {
+            banned[p.operator] = true;
+            emit NodeBanned(p.operator, p.reason, address(this));
+        }
+
+        emit SlashExecuted(
+            proposalId,
+            p.operator,
+            p.reason,
+            p.ticketAmount,
+            p.licenseAmount,
+            p.executed
+        );
+    }
+
+    // ======================
+    // Appeal Functions
+    // ======================
+
+    /// @inheritdoc ISlashingManager
+    function fileAppeal(uint256 proposalId, string calldata evidence) external {
+        require(proposalId < totalProposals, InvalidProposal());
+        SlashProposal storage p = _proposals[proposalId];
+
+        // Only the accused can appeal
+        require(msg.sender == p.operator, Unauthorized());
+        // Only in the window
+        require(block.timestamp < p.executableAt, AppealWindowExpired());
+        // Only once
+        require(!p.appealed, AlreadyAppealed());
+
+        p.appealed = true;
+
+        emit AppealFiled(proposalId, p.operator, p.reason, evidence);
+    }
+
+    /// @inheritdoc ISlashingManager
+    function resolveAppeal(
+        uint256 proposalId,
+        bool appealUpheld,
+        string calldata resolution
+    ) external onlyGovernance {
+        require(proposalId < totalProposals, InvalidProposal());
```

Comment:
do we need this if we have `require(p.appealed, InvalidProposal());` ? if they provide an invalid index, we'd have an empty proposal where appealed would be set to the default boolean value false